### ISSUE TITLE
handle PANIC PANIC PANIC

### ISSUE THREAD

**Issue by nhooyr** [zchee/deoplete-go#45]
If gocode encounters an error, it returns `PANIC PANIC PANIC` but deoplete-go includes that as a completion result. We should instead print the error and return no candidates.

You can test by trying to autocomplete in a empty buffer.

**Comment by zchee** [comment]
@nhooyr Thanks. I'll test later.

**Comment by zchee** [comment]
@nhooyr I tested empty buffer, but cannot reproduce :(
Could you post a sample screenshot or screencast?

**Comment by nhooyr** [comment]
Alright, I forgot how I induced the panic, but if you modify the gocode source to panic, when it does panic (and it will) deoplete-go will show the declaration. Gocode returns a completion with the name `PANIC`, type `PANIC` and class `PANIC`. All we need to do is add a if statement, if the class == `panic` do not display anything.

You can see this in the emacs plugin: https://github.com/nsf/gocode/blob/master/emacs/go-autocomplete.el#L183

**Comment by zchee** [comment]
@nhooyr OK, I can not reproduce, but it is likely to certainly occur.

> name PANIC, type PANIC and class PANIC

I got it. I will consider whether to use which parameters.

**Comment by zchee** [comment]
@nhooyr Added. Find to `PANIC` first letter (`P`)
https://github.com/zchee/deoplete-go/commit/e5ce3ab613c8a5ebde2eeff1ed77cb819540564e

**Comment by nhooyr** [comment]
@zchee you don't have to put it in with the loop. Check if the first one is a panic. The gocode source looks like this:

``` go
    defer func() {
        if err := recover(); err != nil {
            print_backtrace(err)
            c = []candidate{
                {"PANIC", "PANIC", decl_invalid},
            }

            // drop cache
            g_daemon.drop_cache()
        }
    }()
```

It will always be one in length, with the class as panic. (`decl_invalid.String()` returns `PANIC`)

**Comment by zchee** [comment]
@nhooyr Oh, sorry :(
Because I'm not at the desk now, fix again later.

**Comment by nhooyr** [comment]
@zchee alright, i'll create a PR with a fix to this and i'll look into #46

**Comment by zchee** [comment]
@nhooyr Thanks :)
